### PR TITLE
Hubs/Scopes Merge 37 - Fix combined `Contexts`

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

We extended `ConcurrentHashMap`, exposing lots of stuff as public API and making it hard to combine `Contexts`. I've now moved this map into a property we forward to. We can add additional API that was there with the map extension if needed.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by markushi — APPROVED** [review]

File: sentry/src/main/java/io/sentry/protocol/Contexts.java [hunk 1]
```diff
@@ -140,6 +145,69 @@ public void setResponse(final @NotNull Response response) {
     }
   }
 
+  public int size() {
+    return internalStorage.size();
+  }
+
+  public int getSize() {
```
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry/src/main/java/io/sentry/protocol/Contexts.java:152]: It was for allow kotlin to use `.size` without brackets. Is there another way to achieve that?

File: sentry/src/main/java/io/sentry/protocol/Contexts.java [hunk 2]
```diff
@@ -12,16 +12,21 @@
...
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 @Open
-public class Contexts extends ConcurrentHashMap<String, Object> implements JsonSerializable {
+public class Contexts implements JsonSerializable {
   private static final long serialVersionUID = 252445813254943011L;
 
+  private final @NotNull ConcurrentHashMap<String, Object> internalStorage =
+      new ConcurrentHashMap<String, Object>();
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/protocol/Contexts.java:28]: (l) could be `new ConcurrentHashMap<>()`

File: sentry/src/main/java/io/sentry/protocol/Contexts.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry/src/main/java/io/sentry/protocol/Contexts.java:152]: Added some comments to `Contexts` to explain why each is needed.

**Review by lbloder — APPROVED** [review]
LGTM 👍

**Pull request merged**